### PR TITLE
Adding disabled and width attributes to the date-picker

### PR DESCRIPTION
**Pull request by hassananv**

### CODE REVIEW

File: libs/react-components/src/lib/date-picker/date-picker.tsx [hunk 1]
```diff
@@ -24,10 +26,14 @@ export interface GoADatePickerProps extends Margins {
   value?: Date;
   min?: Date;
   max?: Date;
+  disabled?: boolean;
+  width?: string;
   onChange: (name: string, value: Date) => void;
 }
 
 export function GoADatePicker({
+  disabled = false,
```
Comment (chrisolsen) [inline libs/react-components/src/lib/date-picker/date-picker.tsx:35]: The default value will be set within the svelte component, so the `= false` can be removed here.

File: libs/web-components/src/components/date-picker/DatePicker.svelte [hunk 2]
```diff
@@ -14,18 +15,21 @@
...
   export let mb: Spacing = null;
   export let ml: Spacing = null;
+  $: isDisabled = toBoolean(disabled);
 
   let _rootEl: Element;
   let _date: Date;
   let _showPopover: boolean = false;
 
   onMount(async () => {
+    await tick();
```
Comment (chrisolsen) [inline libs/web-components/src/components/date-picker/DatePicker.svelte:32]: Is this fixing an angular binding issue?

File: libs/web-components/src/components/date-picker/DatePicker.svelte [hunk 3]
```diff
@@ -76,7 +80,11 @@
   }
 
   function showCalendar() {
-    _showPopover = true;
+    if (!isDisabled) {
```
Comment (chrisolsen) [inline libs/web-components/src/components/date-picker/DatePicker.svelte:83]: Can you change this to
```
_showPopover = !isDisabled;
```

**Review by chrisolsen — CHANGES_REQUESTED** [review]

File: libs/web-components/src/components/date-picker/DatePicker.svelte (hunk 2 above)
Comment (hassananv) [reply libs/web-components/src/components/date-picker/DatePicker.svelte:32]: Yes, without this line, Angular binding data does not take effect.

File: libs/react-components/src/lib/date-picker/date-picker.tsx [hunk 4]
```diff
@@ -5,6 +5,8 @@ interface WCProps extends Margins {
   ref: React.RefObject<HTMLElement>;
   name?: string;
   value?: string;
+  disabled?: boolean;
```
Comment (chrisolsen) [inline libs/react-components/src/lib/date-picker/date-picker.tsx:8]: The only other thing that is need are some tests.
- validate that the disabled prop is in fact disabling it i.e. doesn't respond to user events (more difficult)
- validate that the width is binding to the internal text input (easy)

**Comment by ArakTaiRoth** [comment]
Re-opened this as we can use this work and just need to add to it.

**Comment by ArakTaiRoth** [comment]
Closing this PR, as part of the work for disabled has been done in this [PR](https://github.com/GovAlta/ui-components/pull/2091)

**Pull request closed without merging**